components/script/dom/htmlimageelement.rs

```diff
@@ -1032,3 +1053,228 @@ fn image_dimension_setter(element: &Element, attr: LocalName, value: u32) {
+                        }
+                        None => {
+                            if !current_descriptor.is_empty() {
+                                descriptors.push_back(current_descriptor.clone());
+                            }
+                            break;
+                        }
+                    }
+                }
+                ParseState::InParens => {
+                    match nextChar {
+                        Some((idx, c @ ')')) => {
+                            current_descriptor.push(c);
+                            state = ParseState::InDescriptor;
+                            continue;
+                        }
+                        Some((_, c)) => {
+                            current_descriptor.push(c);
+                            continue;
+                        }
+                        None => {
+                            if !current_descriptor.is_empty() {
+                                descriptors.push_back(current_descriptor.clone());
+                            }
+                            break;
+                        }
+                    }
+                }
+                ParseState::AfterDescriptor => {
+                    match nextChar {
+                        Some((idx, ' ')) => {
+                            state = ParseState::AfterDescriptor;
+                            continue;
+                        }
+                        Some((idx, c)) => {
+                            state = ParseState::InDescriptor;
+                            buffered = Some((idx, c));
+                            continue;
+                        }
+                        None => {
+                            if !current_descriptor.is_empty() {
+                                descriptors.push_back(current_descriptor.clone());
+                            }
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+        let mut error = Error::No;
+        let mut width: Option<u32> = None;
+        let mut density: Option<f64> = None;
+        let mut future_compat_h: Option<u32> = None;
+        for descriptor in descriptors {
+            let mut char_iter = descriptor.chars();
+            let (digits, remaining) = collect_sequence_characters(&descriptor, is_ascii_digit);
+            let valid_non_negative_integer = parse_unsigned_integer(digits.chars());
+            let has_w = if let "w" = remaining {
+                true
+            } else {
+                false
+            };
+            let valid_floating_point = parse_double(digits);
+            let has_x = if let "x" = remaining {
+                true
+            } else {
+                false
+            };
+            let has_h = if let "h" = remaining {
+                true
+            } else {
+                false
+            };
+            if valid_non_negative_integer.is_ok() && has_w {
```

Comment:
During descriptor parsing, we ensure that there are no repetitions in descriptors, which means there shouldn't be any `w`, `x` or `h` while parsing the values - i.e., if the value ends with `h` and we already have a height value in the descriptor, then it's a parse error.